src/frame/content.rs

```diff
@@ -175,6 +186,51 @@ impl fmt::Display for ExtendedLink {
     }
 }
 
+/// The parsed contents of an general encapsulated object frame.
+#[derive(Clone, Debug, Eq)]
+#[allow(missing_docs)]
+pub struct EncapsulatedObject {
+    pub mime_type: String,
+    pub filename: String,
+    pub description: String,
+    pub data: Vec<u8>,
+}
+
+impl PartialEq for EncapsulatedObject {
+    fn eq(&self, other: &Self) -> bool {
+        self.mime_type == other.mime_type
+            && self.filename == other.filename
+            && self.description == other.description
+    }
+}
+
+impl Hash for EncapsulatedObject {
```

Comment:
If you hash all members you can derive Hash.

But I guess both custom implementations of `Hash` and `PartialEq` should match? Then don't hash `data`.

Reply:
The other implementation apparently don't hash/compare the actual value, just the header data, so I tried to mirror it here as well.

Reply:
That's what I also noticed. If you compare the custom Hash implementations then those also exclude the value whereas you hash (the value) `data`. That's inconsistent.

Reply:
For consistency we must not hash `data`.

Reply:
Does clippy detect such inconsistencies? Otherwise a test could detect them.

Reply:
> For consistency we must not hash `data`.

Fixed.

Reply:
Hash and PartialEq are implemented to act on the primary key (kind of) of the frame. This is typically annotated by the "only one <things>" verses in the spec.

For GEOB frames, the following is specified:

    There may be more than one "GEOB" frame in each tag, but only one with the same content descriptor.

So we should just be hashing the description :) Could you please Hash and PartialEq?